community/gardening/dev-environment.md

```diff
@@ -0,0 +1,222 @@
+    clusters:
+    - cluster:
+        certificate-authority-data: <CAString>
+        server: https://192.168.1.128:6443
+      name: plugin.example.net
+    contexts:
+    - context:
+        cluster: plugin.example.net
+        user: spinnaker-sa
+      name: plugin.example.net
+    current-context: plugin.example.net
+    kind: Config
+    preferences: {}
+    users:
+    - name: spinnaker-sa
+      user:
+        token: <tokenString>
+    ```
+* Configure your Minnaker instance to forward ports to your local machine (gateway) via spinnaker-local.yml configuration.
+  * Create a `.hal/default/profiles/spinnaker-local.yml` file.
+  * Add service settings, or copy settings from `.hal/default/staging/spinnaker.yml` and delete unnecessary services. Read more on [custom service settings](/reference/halyard/custom/#custom-service-settings).
+  * Change the `baseURL` for the service to the default gateway IP Address including the port, e.g. `http://192.168.123.1:8083` for orca.
+  * Consult the [port mappings reference](/reference/architecture/#port-mappings) to determine which ports to forward.
+* Configure the local service to communicate with the Spinnaker instance.
+  * The `kubeconfig` you just created does part of the configuration.
+  * Use `kubectl port-forward` to forward the services required. For example, if running echo locally, you'll need it to communicate with orca and front50:
+      * Option 1: Forward via Service:
+            `kubectl` `--``kubeconfig config-minnaker -n spinnaker port-forward svc/spin-orca 8083:8083 &`
+            `kubectl --kubeconfig config-minnaker -n spinnaker port-forward svc/spin-front50 8080:8080 &`
+      * Option 2: Forward via Pod:
+            `kubectl --kubeconfig config-minnaker -n spinnaker port-forward spin-orca-5f47b76f84-bvh98 8083:8083`
+            `kubectl --kubeconfig config-minnaker -n spinnaker port-forward spin-front50-64ddf796bf-gznqj 8080:8080`
+* __Now you're ready to run and debug the service or services!__
+
+
+
+
+
+## Kubernetes and Docker method:
+
+### Kubernetes installation methods
+_The instructions for this method are in beta. Pull requests welcome!_
+
+* Install Spinnaker to a Kubernetes cluster. There are several ways to do this:
+  * [Install Halyard in Docker](/setup/install/halyard/#install-halyard-on-docker)
+    * In your Docker run command, add a mount the .kube directory to the container to allow you to modify .kube config files on your local machine and persist the changes inside the container:
+    ```
+    docker run -p 8084:8084 -p 9000:9000 \
+      --name halyard --rm \
+      -v ~/.hal:/home/spinnaker/.hal \
+      -v ~/.kube:/home/spinnaker/.kube \
+      -it \
+      gcr.io/spinnaker-marketplace/halyard:stable
+    ```
+  * Consult installation guides for [Google Kubernetes Engine](https://www.spinnaker.io/setup/quickstart/halyard-gke/) and [Amazon Kubernetes Service](https://aws.amazon.com/blogs/opensource/continuous-delivery-spinnaker-amazon-eks/)
```

Comment:
I haven't tested the GKE quickstart in quite a while. Would Spinnaker for GCP be a better choice here than that, @duftler ?

Reply:
I removed that for now after adding the GCE. If we want to re-add later, let's do it.